metrics.nim

```diff
@@ -324,22 +341,28 @@ else:
 
 proc incCounter(counter: Counter, amount: int64|float64 = 1, labelValues: LabelsParam = @[]) =
   when defined(metrics):
-    var timestamp = getTime().toMilliseconds()
+    try:
+      var timestamp = getTime().toMilliseconds()
 
-    if amount < 0:
-      raise newException(ValueError, "Counter.inc() cannot be used with negative amounts.")
+      if amount < 0:
+        raise newException(ValueError, "Counter.inc() cannot be used with negative amounts.")
 
-    let labelValuesCopy = validateCounterLabelValues(counter, labelValues)
+      let labelValuesCopy = validateCounterLabelValues(counter, labelValues)
 
-    atomicAdd(counter.metrics[labelValuesCopy][0].value.addr, amount.float64)
-    atomicStore(cast[ptr int64](counter.metrics[labelValuesCopy][0].timestamp.addr), timestamp.addr, ATOMIC_SEQ_CST)
+      atomicAdd(counter.metrics[labelValuesCopy][0].value.addr, amount.float64)
+      atomicStore(cast[ptr int64](counter.metrics[labelValuesCopy][0].timestamp.addr), timestamp.addr, ATOMIC_SEQ_CST)
 
-    pushMetrics(name = counter.name,
-                value = counter.metrics[labelValuesCopy][0].value,
-                increment = amount.float64,
-                metricType = "c",
-                timestamp = timestamp,
-                sampleRate = counter.sampleRate)
+      pushMetrics(name = counter.name,
+                  value = counter.metrics[labelValuesCopy][0].value,
+                  increment = amount.float64,
+                  metricType = "c",
+                  timestamp = timestamp,
+                  sampleRate = counter.sampleRate)
+    except Exception as e:
+      try:
+        write stderr, "metrics: " & e.msg
+      except IOError:
+        discard
```

Comment:
Why hide IOError?

Reply:
to avoid a crash, for example if someone is running as daemon and closed stderr for whatever reason

Reply:
Why would I want to handle that scenario in a library and why would I want to handle it by hiding it?

Reply:
we're doing a best effort here - in the future, more sophisticate features could be added

Reply:
No future plans can excuse a drop in quality for the API user right now.

Reply:
as a user of the library, I consider this an improvement, really - because there is nothing meaningful I can do with this error in the spot that I'm incrementing the counter - the code that increments counters is never equipped to handle errors in metrics due to the nature of metrics.

We could potentially raise a Defect here - that still does a disservice to our users because the smallest programmer mistake will crash the application for a library that people normally consider an optional add-on